tests/run/t6260-delambdafy.scala

```diff
@@ -0,0 +1,17 @@
+class C[A](private val a: Any) extends AnyVal
+
+object Test {
+  val f = (x: C[Any]) => {println(s"f($x)"); x}
+  def main(args: Array[String]): Unit = {
+    f(new C("."))
+
+    // Lambdas in Dotty are created with LambdaMetaFactory (when possible)
+    // which does not guarantee that the list of methods will always be the same.
+    /*
+    val methods = f.getClass.getDeclaredMethods.map(_.getName).sorted
```

Comment:
This test tests what happens in this particular place: that lambda is created. It wouldn't be created if method required an outer-pointer. It should continue to list methods, but check file should not contain `Test$lambda$1$$apply` anymore.